src/segments/umbraco.go

```diff
@@ -0,0 +1,231 @@
+	// OR one or more .csproj files at the same level as the Umbraco folder
+
+	// Modern .NET Core based Umbraco
+	if findUmbracoResults.FoundCSProj {
+		u.env.Debug("UMBRACO: Checking for modern Umbraco as we have found .csproj file")
+		return u.TryFindModernUmbraco(findUmbracoResults.FilePath)
+	}
+
+	// Legacy .NET Framework based Umbraco
+	if findUmbracoResults.FoundWebConfig {
+		u.env.Debug("UMBRACO: Checking for legacy Umbraco as we have found web.config file")
+		return u.TryFindLegacyUmbraco(findUmbracoResults.FilePath)
+	}
+
+	// If we have got here then neither modern or legacy Umbraco was NOT found
+	u.FoundUmbraco = false
+	return false
+}
+
+func (u *Umbraco) Template() string {
+	return "{{.Version}} "
+}
+
+func (u *Umbraco) Init(props properties.Properties, env platform.Environment) {
+	u.props = props
+	u.env = env
+}
+
+func (u *Umbraco) TryFindUmbracoInParentDirsOrSelf() (*FindUmbracoResult, error) {
+	defer u.env.Trace(time.Now(), "UMBRACO: Checking for an umbraco folder & files in current or any parent folders until root")
+	currentFolder := u.env.Pwd()
+	results := FindUmbracoResult{}
+
+	for {
+		// Check if a directory named "Umbraco" exists in the current directory
+		files := u.env.LsDir(currentFolder)
+
+		// Have to loop over each item found in the current folder
+		for _, file := range files {
+			// Check if the item is a folder AND it matches 'Umbraco' regardless of casing
+			if file.IsDir() && strings.EqualFold(file.Name(), umbracoFolderName) {
+				u.env.Debug("UMBRACO: Found an Umbraco folder in " + currentFolder)
+				results.FoundUmbracoFolder = true
+			}
+
+			// Check if the item is a file AND it matches 'web.config' regardless of casing
+			if !file.IsDir() && strings.EqualFold(file.Name(), umbracoWebConfig) {
+				u.env.Debug("UMBRACO: Found a web.config file in " + currentFolder)
+				results.FoundWebConfig = true
+				results.FilePath = filepath.Join(currentFolder, file.Name())
+			}
+
+			// Check if the item is a file AND has file extension of .csproj regardless of casing
+			if !file.IsDir() && strings.EqualFold(filepath.Ext(file.Name()), ".csproj") {
+				u.env.Debug("UMBRACO: Found a .csproj file in " + currentFolder)
+				results.FoundCSProj = true
+				results.FilePath = filepath.Join(currentFolder, file.Name())
+			}
+
+			// If we have found an Umbraco folder AND a .csproj OR  an Umbraco folder AND a web.config file
+			// Then we can break the for loop as we have found what we need
+			if (results.FoundUmbracoFolder && results.FoundCSProj) || (results.FoundUmbracoFolder && results.FoundWebConfig) {
+				// Break out the loop for checking the collection of files in the current folder
+				break
+			}
+		}
+
+		// If we have found an Umbraco folder AND a .csproj OR  an Umbraco folder AND a web.config file
+		// Then we can break the for loop as we have found what we need
+		if (results.FoundUmbracoFolder && results.FoundCSProj) || (results.FoundUmbracoFolder && results.FoundWebConfig) {
```

Comment:
you could use a `found := false` above which is set when all condition match during search as it's the same logic as to decide when to break the for loop above.